[WIP] Update dev environment to better represent production.

whoodes opened the pull request:
This pull request is an effort to bridge the gap between development and production environments.
The preliminary steps include two parts:

- [ ] Migrate the codebase to use the variable SYSTEM_BOT_REALM in places where we mean that instead of 'zulip'.

- [x] Change SYSTEM_BOT_REALM in ```dev_settings.py``` to 'zulipinternal' and have the system bots for that realm created using the same code as ```zerver/management/commands/initialize_voyager_db```.

The thread relating to this pull request in the Zulip community can be found [here](https://chat.zulip.org/#narrow/stream/49-development-help/topic/realms.20from.20populate_db/near/694068).

(The following steps are TBD.)

whoodes commented:
Hi @timabbott  , I have been spending some time  exploring different avenues of code from the failing tests as a result of moving the system bots. I was seeing a lot pickling errors such as, ```PicklingError("Can't pickle <class 'zerver.models.DoesNotExist'>:
    attribute lookup DoesNotExist on zerver.models failed",)```, (wanted to make sure that this is something to be expected).  I was diving pretty deep just tracing errors, but I wanted to be sure I am reasonable with my approach to moving the system bots before committing to any further changes.
(I'm not sure if this merited a review, or was something that should have been asked in chat.)

Thank you.

timabbott commented on zilencer/management/commands/populate_db.py:
```diff
@@ -209,27 +209,53 @@ def handle(self, **options: Any) -> None:
+            # Create realm for system bots, logic drawn from initialize_voyager_db.
+            zulipinternal_realm = Realm.objects.create(string_id=settings.SYSTEM_BOT_REALM,
+                                                       name='System Bots')
+            RealmDomain.objects.create(realm=zulipinternal_realm, domain='zulipinternal.com')
```
we likely don't want a RealmDomain for this.

timabbott commented:
I don't know what would cause such a PicklingError; I'd need to see some code causing it to help.

whoodes commented:
Looks like the pickling was being caused by the welcome bot not being found.  I added it into ```dev_settings```, and removed the ```Realm.Domain``` line.  I was running up against miscounts with the ```cross_realm_bots```, so I moved new user bot into 'zulipinternal', along with error and default bots. This seems to set everything up nicely for the next step.  (Also named the filler bots more appropriately.)

whoodes commented:
Closing this as the pull request is no longer relevant.
